storage-proofs/src/zigzag_drgporep.rs

```diff
@@ -1,46 +1,1066 @@
+                let mut exp_parents_odd_proofs = Vec::with_capacity(challenges.len());
+
+                // ZigZag commitment specifics
+                for raw_challenge in challenges {
+                    let challenge = raw_challenge % graph_1.size();
+
+                    // Initial data layer openings (D_X in Comm_D)
+                    {
+                        comm_d_proofs.push(DataProof {
+                            data: aux.tree_d.read_at(challenge),
+                            proof: MerkleProof::new_from_proof(&aux.tree_d.gen_proof(challenge)),
+                        });
+                    }
+
+                    // C_n-X+1 in Comm_C
+                    {
+                        comm_c_proofs_even.push(MerkleProof::new_from_proof(
+                            &aux.tree_c.gen_proof(inv_index(nodes_count, challenge)),
+                        ));
+                    }
+
+                    // C_X in Comm_C
+                    {
+                        comm_c_proofs_odd.push(MerkleProof::new_from_proof(
+                            &aux.tree_c.gen_proof(challenge),
+                        ));
+                    }
+
+                    // e_X^(j) and e_n-X+1^(j)
+                    {
+                        let mut labels = Vec::with_capacity(2 * layers);
+                        for layer in 1..layers {
+                            // odd
+                            labels.push(
+                                // -1 because encodings is zero indexed
+                                get_node::<H>(&tau.encodings[layer - 1], challenge)?,
+                            );
+
+                            // even
+                            labels.push(get_node::<H>(
+                                // -1 because encodings is zero indexed
+                                &tau.encodings[layer - 1],
+                                inv_index(nodes_count, challenge),
+                            )?);
+                        }
+                        layer_labels.push(labels);
+                    }
+
+                    // Final replica layer openings (e_n-X-1^(l))
+                    {
+                        let challenge_inv = inv_index(nodes_count, challenge);
+                        comm_r_last_proofs.push(DataProof {
+                            data: aux.tree_r_last.read_at(challenge_inv),
+                            proof: MerkleProof::new_from_proof(
+                                &aux.tree_r_last.gen_proof(challenge_inv),
+                            ),
+                        });
+                    }
+
+                    // DRG Parents
+                    {
+                        let base_degree = graph_1.base_graph().degree();
+
+                        // DRG.Parents(X, 1)
+                        let mut drg_parents = vec![0; base_degree];
+                        graph_1.base_parents(challenge, &mut drg_parents);
```

Comment:
Let's make sure that `base_parents` includes all of the DRG parents, including the immediate predecessor. (I'm not closely tracking the issue which might add this.)